### PR TITLE
Added output-directory for better flexibility

### PR DESCRIPTION
**Pull request by MohammedAtif**
Reference : https://docs.fastlane.tools/actions/gym/#parameters

> Not backward compatible to avoid ambiguity
P.S. I use this action for React Native and without correct output directory managing build and distribution is difficult

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -58,7 +58,8 @@ platform :ios do
         workspace: ENV["WORKSPACE_PATH"],
         configuration: ENV["CONFIGURATION"],
         scheme: ENV["SCHEME"],
-        output_name: ENV["OUTPUT_PATH"],
+        output_directory = ENV["OUTPUT_DIRECTORY"],
+        output_name: ENV["OUTPUT_NAME"],
```
Comment (yukiarrr) [inline fastlane/Fastfile:62]: I think `output-path` is simpler for users who don't know fastlane.
Also, some users may write `use: yukiarrr/ios-build-action@master` when using `ios-build-action`.
So I think we should leave `output-path` as it is and split it internally into a directory name and a file name.

```suggestion
        output_directory: File.dirname(ENV["OUTPUT_PATH"]),
        output_name: File.basename(ENV["OUTPUT_PATH"]),
```

**Review by yukiarrr — CHANGES_REQUESTED** [review]
@MohammedAtif Thanks for your contribution! One request.

File: fastlane/Fastfile (hunk 1 above)
Comment (MohammedAtif) [reply fastlane/Fastfile:62]: @yukiarrr This sounds good. Considering backward compatibility, this can be done too.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -58,7 +58,8 @@ platform :ios do
         workspace: ENV["WORKSPACE_PATH"],
         configuration: ENV["CONFIGURATION"],
         scheme: ENV["SCHEME"],
-        output_name: ENV["OUTPUT_PATH"],
+        output_directory = File.dirname(ENV["OUTPUT_PATH"]),
```
Comment (yukiarrr) [inline fastlane/Fastfile:61]:
```suggestion
        output_directory: File.dirname(ENV["OUTPUT_PATH"]),
```

File: fastlane/Fastfile [hunk 3]
```diff
@@ -67,7 +68,8 @@ platform :ios do
         project: ENV["PROJECT_PATH"],
         configuration: ENV["CONFIGURATION"],
         scheme: ENV["SCHEME"],
-        output_name: ENV["OUTPUT_PATH"],
+        output_directory = File.dirname(ENV["OUTPUT_PATH"]),
```
Comment (yukiarrr) [inline fastlane/Fastfile:71]:
```suggestion
        output_directory: File.dirname(ENV["OUTPUT_PATH"]),
```

**Review by yukiarrr — CHANGES_REQUESTED** [review]
@MohammedAtif Thanks for the fix!
Can you fix the following to match the syntax?

**Review by yukiarrr — APPROVED** [review]
Thanks for the very fast fix!
In a few days, I will release it as v1.2.0, including this change.

**Comment by yukiarrr** [comment]
@MohammedAtif Including this PR, v1.2.0 has been released.

**Comment by MohammedAtif** [comment]
Thank @yukiarrr for the update 👍

**Pull request merged**